modules/library/referencing/src/main/java/org/geotools/referencing/factory/ManyAuthoritiesFactory.java

```diff
@@ -166,33 +168,45 @@ private static Collection<AuthorityFactory> createFallbacks(
          * the collection of factories for each authority.
          */
         int authorityCount = 0;
-        final Citation[] authorities = new Citation[factories.size()];
-        @SuppressWarnings("unchecked")
-        final List<AuthorityFactory>[] factoriesByAuthority = new List[authorities.length];
+        final List<Citation> authorities = new ArrayList<>(factories.size());
```

Comment:
My my this method is painful to read... what would be needed here is a MultiMap with a custom equality, but there is no multimap in the classpath.

Anyhow the changes seem legit...

Reply:
Yes, the method is painful. Haven't looked at MultiMap, but I do believe it could be written more compact.